Bottom: 58e09cfee9b2a170f6a54383a695eb66ea8c82a5
Top:    23e8cf6e90590e8e7a529d4aa9aa7f9111967f7a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-28 15:43:46 +0100

draw.c: Move doing_assy flag assignment into PrintAssembly()

Keeps things tidier


---

diff --git a/src/draw.c b/src/draw.c
index 7d7a594..ea3e1ce 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -639,6 +639,7 @@ PrintAssembly (int side, const BoxType * drawn_area)
 {
   int side_group = GetLayerGroupNumberByNumber (max_copper_layer + side);
 
+  doing_assy = true;
   gui->set_draw_faded (Output.fgGC, 1);
   DrawLayerGroup (side_group, drawn_area);
   DrawPPV (side_group, drawn_area);
@@ -646,6 +647,7 @@ PrintAssembly (int side, const BoxType * drawn_area)
 
   /* draw package */
   DrawSilk (side, drawn_area);
+  doing_assy = false;
 }
 
 /* ---------------------------------------------------------------------------
@@ -787,8 +789,6 @@ DrawEverything (BoxTypePtr drawn_area)
       gui->end_layer ();
     }
 
-  doing_assy = true;
-
   if (gui->set_layer ("topassembly", SL (ASSY, TOP), 0))
     {
       PrintAssembly (COMPONENT_LAYER, drawn_area);
@@ -801,8 +801,6 @@ DrawEverything (BoxTypePtr drawn_area)
       gui->end_layer ();
     }
 
-  doing_assy = false;
-
   if (gui->set_layer ("fab", SL (FAB, 0), 0))
     {
       PrintFab (Output.fgGC);
